### PR TITLE
Add 'Index.list_documents' API wrapper.

### PR DESCRIPTION
**Pull request by tseaver**

### CODE REVIEW

File: gcloud/search/test_client.py [hunk 1]
```diff
@@ -54,10 +54,10 @@ def test_list_indexes_defaults(self):
         client = self._makeOne(self.PROJECT, creds)
         conn = client.connection = _Connection(DATA)
 
-        zones, token = client.list_indexes()
+        indexes, token = client.list_indexes()
```
Comment (dhermes) [inline gcloud/search/test_client.py]: This was just copy-pasta?

File: gcloud/search/test_index.py [hunk 2]
```diff
@@ -105,9 +105,168 @@ def test_from_api_repr_w_properties(self):
         self.assertTrue(index._client is client)
         self._verifyResourceProperties(index, RESOURCE)
 
+    def test_list_documents_defaults(self):
```
Comment (dhermes) [inline gcloud/search/test_index.py]: These tests always feel so long. Do you think there's anything we can do to make them shorter?

File: gcloud/search/test_client.py (hunk 1 above)
Comment (tseaver) [inline gcloud/search/test_client.py]: Yup.

File: gcloud/search/test_index.py (hunk 2 above)
Comment (tseaver) [inline gcloud/search/test_index.py]: d8961e5 factors out the common document setup / verification code from the two `list_documents` testcase methods.

**Comment by dhermes** [comment]
LGTM

**Pull request merged**